Meadowlark-js/packages/metaed-plugin-edfi-meadowlark/src/enhancer/HasSchoolYearEnhancer.ts

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { EnhancerResult, MetaEdEnvironment, getAllEntitiesOfType, ModelBase } from '@edfi/metaed-core';
+import { CollectedProperty } from '../model/CollectedProperty';
+import { EntityMeadowlarkData } from '../model/EntityMeadowlarkData';
+
+const enhancerName = 'HasSchoolYearEnhancer';
+
+/**
+ * This enhancer  pre-determines if a given entity type contains a schoolYearEnumeration.
+ */
+export function enhance(metaEd: MetaEdEnvironment): EnhancerResult {
+  getAllEntitiesOfType(
+    metaEd,
+    'domainEntity',
+    'association',
+    'domainEntitySubclass',
+    'associationSubclass',
+    'common',
+  ).forEach((entity: ModelBase) => {
+    const meadowlarkEntity = entity.data.meadowlark as EntityMeadowlarkData;
+    meadowlarkEntity.hasSchoolYear = meadowlarkEntity.collectedProperties.some(
+      (property: CollectedProperty) => property.property.type === 'schoolYearEnumeration',
+    );
+  });
```

Comment:
Is this the right list of entity types?

Note: this does not dive into reference types. That concerned me momentarily... but then I realized we really only need to validate any `schoolYearReference` directly on the object. Reference validation will handle any nested references.

For example, a bad year in `Session` keeps that Session from being saved.

A bad year in `CourseOffering.Session` will not be rejected by the Year validation - but it _will_ be rejected because the referenced ` Session` does not exist.